content/blog/wireshark-protobuf-and-grpc-dissectors.md

```diff
@@ -0,0 +1,243 @@
+![wireshark_protobuf_search_paths](/img/wireshark_protobuf_search_paths.png)
+This preferences dialog can be found by menu 'Edit->Preferences->Protocols->ProtoBuf->Protobuf search paths'.
+
+### gRPC Sample Capture files
+
+There are two gRPC sample capture files [grpc_person_search_protobuf_with_image.pcapng](https://gitlab.com/wireshark/wireshark/-/wikis/uploads/f6fcdceb0248669c0b057bd15d45ab6f/grpc_person_search_protobuf_with_image.pcapng) and [grpc_person_search_json_with_image.pcapng](https://gitlab.com/wireshark/wireshark/-/wikis/uploads/88c03db83efb2e3253c88f853d40477b/grpc_person_search_json_with_image.pcapng) on the [SampleCaptures page](https://gitlab.com/wireshark/wireshark/-/wikis/SampleCaptures) of Wireshark. The difference between the two captures is that the message of former is serialized in Protobuf, and the latter is serialized in JSON.
+
+To decode sample captures as gRPC message, you must decode the traffic on TCP port 50051 and 50052 as HTTP2:
+![wireshark_decode_as_dialog](/img/wireshark_decode_as_dialog.png)
+
+#### gRPC Capture serialized in Protobuf
+
+This is a screenshot of the gRPC service request message of [grpc_person_search_protobuf_with_image.pcapng](https://gitlab.com/wireshark/wireshark/-/wikis/uploads/f6fcdceb0248669c0b057bd15d45ab6f/grpc_person_search_protobuf_with_image.pcapng):
+![wireshark_grpc_protobuf_search_request](/img/wireshark_grpc_protobuf_search_request.png)
+That shows searching for the Person objects of Jason and Lily based on their names. 
+
+Since the `Search` operation is defined as the server streaming RPC mode, the Person objects can be return back to client one after another:
+![wireshark_grpc_protobuf_search_response](/img/wireshark_grpc_protobuf_search_response.png)
+
+You may have noticed that the `portrait_image` of bytes type is parsed into a PNG data. It's because we register the PNG dissector in the `"protobuf_field"` dissector table for parsing the value of field `portrait_image` as picture by putting following Lua script `'protobuf_portrait_field.lua'` into the 'plugins' subdirectory of 'Personal configuration' directory:
+
+```lua
+do
+    local protobuf_field_table = DissectorTable.get("protobuf_field")
+    local png_dissector = Dissector.get("png")
+    protobuf_field_table:add("tutorial.Person.portrait_image", png_dissector)
+end
+```
+
+#### gRPC Capture serialized in JSON
+
+Following is a screenshot of [grpc_person_search_json_with_image.pcapng](https://gitlab.com/wireshark/wireshark/-/wikis/uploads/88c03db83efb2e3253c88f853d40477b/grpc_person_search_json_with_image.pcapng), which serializes gRPC message with JSON:
+![wireshark_grpc_json_search_response](/img/wireshark_grpc_json_search_response.png)
+
+>Note, if the gRPC message is serialized in JSON, the *'Protobuf Search Paths'* does not need to be configured.
+
+### Protobuf Sample Capture files
+
+#### Built-in Protobuf UDP dissector
```

Comment:
I don't think this UDP and custom dissectors fits all that well in this post. It is substantially more complicated and I'm worried that it will scare off users thinking the normal flow is complex. The large lua snippet also doesn't seem all that well suited for a blog post. (I can tell it isn't actually _that_ complicated, but it seems complex and is more than users need.)

That said, it looks like it is good content. So if this content isn't elsewhere, it seems it should find a home. There's a small chance it could make sense to have a "Advanced Protobuf" themed post talking about this, since Protobuf is important to the gRPC community. But raw UDP and TCP aren't all that important to the gRPC community, so it seems a weak fit.

Reply:
I have removed these sections, and instead leaving a link in "Read more about Protobuf Dissector" section.